frontend/model/state.js

```diff
@@ -329,6 +335,235 @@ const getters = {
+            //         month's paymentsFrom.
+            // Let C = the total amount we "owe" to payment.to from the
+            //         current month's distribution.
+            // Let D = the amount we're pledging this month
+            // Let E = the amount still unpaid for the previous month's distribution,
+            //         calculated as: C > 0 ? A : A + D - B
+            //
+            // If E > 0, then display a row for the late payment.
+            const A = payment.amount
+            const B = this.paymentTotalFromUserToUser(this.ourUsername, payment.to, cMonthstamp)
+            var C = currentDistribution
+              .filter(a => a.from === payment.from && a.to === payment.to)
+            C = C.length > 0 ? C[0].amount : 0
+            const D = pledgeAmount
+            const E = C > 0 ? A : A + D - B
+            if (E > 0) {
+              latePayments.push({
+                username: payment.to,
+                displayName: this.userDisplayName(payment.to),
+                amount: payment.amount, // TODO: include currency (what if it was changed?)
+                // partiaL: TODO add this info as it is in this.paymentsTodo
+                isLate: true,
+                date: dueIn
+              })
+            }
+          }
+        }
+        */
+
+        // On the other hand this seems to work as expected. "partial" is the only thing missing
+        for (const payment of adjusted) {
+          if (payment.from !== ourUsername) {
+            continue
+          }
+
+          latePayments.push({
+            ...payment,
+            // partial: TODO add this info, but what's the cleanest way to do it?
+            date: dueIn
+          })
+        }
+      }
+      return latePayments
+    })()
+    const toBeReceived = (() => {
+      const unadjusted = getters.groupIncomeDistribution.filter(p => p.to === ourUsername)
+      const payments = []
+      for (const p of getters.groupIncomeAdjustedDistribution) {
+        if (p.to !== ourUsername) { continue }
+
+        const existPayment = unadjusted.find(({ from }) => from === p.from) || { amount: 0 }
+        const amount = +currency.displayWithoutCurrency(p.amount)
+        const existingAmount = +currency.displayWithoutCurrency(existPayment.amount)
+
+        if (amount <= 0) { continue }
+
+        const partialAmount = existingAmount - amount
+        const existingPayment = {}
+        if (partialAmount > 0) {
+          // TODO/BUG this only work if the payment is done in 2 parts. if done in >=3 won't work.
+          const receivePartial = received.find((r) => r.username === p.to && r.amount === partialAmount)
+          if (receivePartial) {
+            existingPayment.hash = receivePartial.hash
+          }
+        }
+        payments.push({
+          ...existingPayment,
+          ...p,
+          total: existingAmount,
+          partial: partialAmount > 0
+        })
+      }
+      return payments
```

Comment:
Hmm. This part I still need to review since this is new. Don't quite fully understand yet. Maybe, could you add a comment above line 492 explaining why you found it necessary to implement this?

Reply:
We need to know what payments are expected to be received, to be displayed at `monthOverview` (the sidebar) on the view of a receiver. [see figma screen](https://www.figma.com/file/mxGadAHfkWH6qApebQvcdN/Group-Income-2.0?node-id=9518%3A21)